element.h

```diff
@@ -3,6 +3,15 @@
 
 #include <vector>
 
+/**
+ * Contains the elements used, as well as any
+ * categories of elements.
```

Comment:
It isn't clear to me what this means, and I don't think 'contains' is the best word here. What about something like:
```suggestion
 * Element defines constants for each element (5 elements) and
 * constants for each useful combination (e.g.: ALL).
```